pkg/controller/compute/managed.go

```diff
@@ -143,14 +146,42 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, errors.New(errNotAKSCluster)
 	}
 	cr.SetConditions(xpv1.Creating())
-	secret, err := e.newPasswordFn()
+
+	pw, err := e.getPassword(ctx, cr)
 	if err != nil {
-		return managed.ExternalCreation{}, errors.Wrap(err, errGenPassword)
+		return managed.ExternalCreation{}, err
+	}
+	if pw == "" {
+		pw, err = e.newPasswordFn()
+		if err != nil {
+			return managed.ExternalCreation{}, errors.Wrap(err, errGenPassword)
+		}
+	}
+	return managed.ExternalCreation{
+		ConnectionDetails: managed.ConnectionDetails{
+			xpv1.ResourceCredentialsSecretPasswordKey: []byte(pw),
+		},
+	}, errors.Wrap(e.client.EnsureManagedCluster(ctx, cr, pw), errCreateAKSCluster)
+}
+
+func (e *external) getPassword(ctx context.Context, cr *v1alpha3.AKSCluster) (string, error) {
```

Comment:
nitpick: in [RDS instance](https://github.com/crossplane/provider-aws/blob/0fab933/pkg/controller/rds/dbinstance/setup.go#L90), we covered the auto-generate scenario by always asking for a secret but then having the controller populate it with the auto-generated one so that the flow of sensitive information is always the same.